src/plugins/content_management/public/registry/registry.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { ContentTypeDetails } from './content_type_details';
+import { ContentType } from './content_type';
+
+export class ContentTypeRegistry {
+  private readonly types: Map<string, ContentType> = new Map();
```

Comment:
Follow up on my comment about naming: this would then become `private readonly contents`. Which would map to the same property on the server (https://github.com/elastic/kibana/pull/150692/files#diff-193771255be169262e0e94375866337b08dcc2c7e167702faf20c825e7f2b678R14)